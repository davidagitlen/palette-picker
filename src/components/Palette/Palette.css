.Palette {
  border: 1px solid #c8c8c8;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  height: 50px;
  width: 100%;
  box-shadow: 3px 3px 7px #A9A9A9;
}

.Palette:hover {
  border: 1px solid #424A60;
  cursor: pointer;
}

.palette-name-wrapper {
  background-color: snow;
  height: 100%;
  padding: 0 0 0 5px;
  width: 45%;
}

.palette-name-wrapper > h3 {
  font-size: 18px;
}

.swatch-wrapper {
  display: flex;
  flex-direction: row;
  height: 100%;
  width: 45%;
}

.swatches {
  height: 100%;
  width: 20%;
}

.trash {
  height: 30px;
  margin: auto;
  width: 30px;
}