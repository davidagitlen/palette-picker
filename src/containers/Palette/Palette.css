.Palette {
  display: flex;
  flex-direction: row;
  height: 50px;
  width: 100%;
  border: 1px solid black;
}

.palette-name-wrapper {
  height: 100%;
  width: 50%;
}

.swatch-wrapper {
  display: flex;
  flex-direction: row;
  height: 100%;
  width: 50%;
}

.swatches {
  height: 100%;
  width: 20%;
}

.trash {
  height: 35px;
  width: 35px;
  display: flex;
  flex-direction: row;
}