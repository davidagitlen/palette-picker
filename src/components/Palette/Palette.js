import React from "react";
import "./Palette.css";
import trash from '../../images/trash.svg';

const Palette = ({palette, editPalette, trashPalette, selectPalette}) => {
  const {
    hex_1,
    hex_2,
    hex_3,
    hex_4,
    hex_5
  } = palette;
  const hexArray = [hex_1, hex_2, hex_3, hex_4, hex_5];
  const swatchList = hexArray.map((hex, i) => 
    <div 
      className='swatches'
      key={i + Date.now()}
      style={{ backgroundColor: hex}}
    >
    </div>
    )
  return (
    <>
    <article className="Palette">
      <div 
        className='palette-name-wrapper'
        onClick={() => selectPalette(palette)}
      >
        <h3>{palette.palette}</h3>
      </div>
      <div 
        className='swatch-wrapper'
        onClick={() => selectPalette(palette)}
        >
        {swatchList}
      </div>
      <input
        className='trash'
        type='image'
        alt='trash'
        src={trash}
        onClick={() => trashPalette(palette.id)}
      />
    </article>
    </>
  );
};

export default Palette;
