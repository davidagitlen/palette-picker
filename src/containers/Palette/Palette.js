import React from "react";
import "./Palette.css";
import update from '../../images/update.svg';
import trash from '../../images/trash.svg';

const Palette = ({palette, editPalette, trashPalette}) => {

  return (
    <article className="Palette">
      <h3>Palette</h3>
    </article>
  );
};

export default Palette;
