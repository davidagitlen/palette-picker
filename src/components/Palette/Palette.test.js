import React from 'react';
import { shallow } from 'enzyme';
import Palette from './Palette';

it('initial test to pass CI suite', () => {
  expect(true).toEqual(true);
});

describe('Palette', () => {

  let wrapper, palette, mockEditPalette, mockTrashPalette, mockSelectPalette;

  beforeEach(() => {

    palette = {
      hex_1: "#3192EC",
      hex_2: "#77B8AB",
      hex_3: "#910E31",
      hex_4: "#D8C86A",
      hex_5: "#6E16C2",
      id: 34,
      palette: "one",
      project_id: 1,
      project_name: "Blah blah"
    };
    mockEditPalette = jest.fn();
    mockTrashPalette = jest.fn();
    mockSelectPalette = jest.fn();
    global.Date = {
      now: jest.fn().mockImplementation(() => 12345)
    };
    wrapper = shallow( 
      <Palette 
        palette={palette}
        editPalette={mockEditPalette}
        trashPalette={mockTrashPalette}
        selectPalette={mockSelectPalette}
      />
    );
  });

  it('should match the snapshot with data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke selectPalette with the current palette passed in when palette name is clicked', () => {

    wrapper.find('.palette-name-wrapper').simulate('click');

    expect(mockSelectPalette).toHaveBeenCalledWith(palette);
  });

  it('should invoke selectPalette with the current palette passed in when swatch is clicked', () => {

    wrapper.find('.swatch-wrapper').simulate('click');

    expect(mockSelectPalette).toHaveBeenCalledWith(palette);
  });

  it('should invoke trashPalette with the current palette\'s id passed in an as argument when the trash svg is clicked', () => {

    wrapper.find('.trash').simulate('click');

    expect(mockTrashPalette).toHaveBeenCalledWith(palette.id);
  });
});