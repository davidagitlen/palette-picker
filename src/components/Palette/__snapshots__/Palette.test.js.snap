// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match the snapshot with data passed in correctly 1`] = `
<Fragment>
  <article
    className="Palette"
  >
    <div
      className="palette-name-wrapper"
      onClick={[Function]}
    >
      <h3>
        one
      </h3>
    </div>
    <div
      className="swatch-wrapper"
      onClick={[Function]}
    >
      <div
        className="swatches"
        key="12345"
        style={
          Object {
            "backgroundColor": "#3192EC",
          }
        }
      />
      <div
        className="swatches"
        key="12346"
        style={
          Object {
            "backgroundColor": "#77B8AB",
          }
        }
      />
      <div
        className="swatches"
        key="12347"
        style={
          Object {
            "backgroundColor": "#910E31",
          }
        }
      />
      <div
        className="swatches"
        key="12348"
        style={
          Object {
            "backgroundColor": "#D8C86A",
          }
        }
      />
      <div
        className="swatches"
        key="12349"
        style={
          Object {
            "backgroundColor": "#6E16C2",
          }
        }
      />
    </div>
    <input
      alt="trash"
      className="trash"
      onClick={[Function]}
      src="trash.svg"
      type="image"
    />
  </article>
</Fragment>
`;
